larq/quantized_variable.py

```diff
@@ -0,0 +1,308 @@
+"""Contains QuantizedVariable, a variable that can be quantized in the forward pass."""
+import tensorflow as tf
+from tensorflow.python.distribute.values import DistributedVariable
+from tensorflow.python.framework import ops
+from tensorflow.python.ops import resource_variable_ops
+
+from larq import quantized_scope
+
+
+class QuantizedVariable(tf.Variable):
+    """A Variable that can be quantized in the forward pass in applicable contexts."""
+
+    def __init__(self, variable, quantizer=None, precision=None):
+        """Creates an QuantizedVariable instance.
+
+        # Arguments
+        variable: A floating-point resource variable to wrap.
+        quantizer: An optional quantizer to transform the floating-point
+            variable to a fake quantized variable.
+        precision: An optional integer defining the precision of the quantized
+            variable. If `None`, `quantizer.precision` is used.
+        """
+        if not resource_variable_ops.is_resource_variable(variable):
+            raise ValueError(
+                f"variable must be of type tf.ResourceVariable, but got: {variable}"
```

Comment:
Do we want to report `type(variable)` here?

Reply:
Yeah might be a good thing. I only copied this input validation from `AutoCastVariable` without thinking to much about it ;)